### ISSUE TITLE
MSS doesn't close when "quit"ing while being connected to MSCO server

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#490]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/490) by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

In two instances, MSS didn’t close upon selecting “quit” from the main menu. 

logfile is being filled further with messages like:  

```

t='localhost', port=8083): Max retries exceeded with url: /socket.io/?transport=polling&EIO=3&t=1572344160.6430178 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe788d0b320>: Failed to establish a new connection: [Errno 111] Connection refused')).
2019-10-29 11:16:00,647 (client._handle_reconnect:535): Connection failed, new attempt in 5.49 seconds
2019-10-29 11:16:06,142 (client._connect_polling:269): Attempting polling connection to http://localhost:8083/socket.io/?transport=polling&EIO=3
2019-10-29 11:16:06,145 (connectionpool._new_conn:205): Starting new HTTP connection (34): localhost:8083
2019-10-29 11:16:06,146 (client._send_request:454): HTTP GET request to http://localhost:8083/socket.io/?transport=polling&EIO=3&t=1572344166.1431236 failed with error HTTPConnectionPool(host='localhost', port=8083): Max retries exceeded with url: /socket.io/?transport=polling&EIO=3&t=1572344166.1431236 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe78a734710>: Failed to establish a new connection: [Errno 111] Connection refused')).
2019-10-29 11:16:06,146 (client._handle_reconnect:535): Connection failed, new attempt in 5.05 seconds
```

Maybe because I killed the local MSCO server first?

**Comment by ReimarBauer** [comment]
**Original comment by Shivashis Padhi (Bitbucket: [plant99](https://bitbucket.org/plant99), GitHub: [plant99](https://github.com/plant99)).**

----------------------------------------

This is because the client keeps trying to connect to socket-server, I’ll see if there is a timeout option to it.

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

I can connect and login to msco without there being a problem.

As soon as I double click a project, MSS doesn’t quit anymore.

‌

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

OK, I digged a bit into this. 

If I close the mscolab window, the socketio background process is terminated after a couple of seconds.

If I close the main window, the socketio process is never disconnected. I believe that we need to shutdown the mscolab window manually from the main process similar to how the views are handled.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Merged in plant99/mss/i490 (pull request #753)

fixes issue#490
